api/v1/nodeinventory/service.go

```diff
@@ -0,0 +1,59 @@
+package nodeinventory
+
+import (
+	"context"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	apiGRPC "github.com/stackrox/scanner/api/grpc"
+	v1 "github.com/stackrox/scanner/generated/scanner/api/v1"
+	"github.com/stackrox/scanner/pkg/nodeinventory"
+	"google.golang.org/grpc"
+)
+
+// Service defines the node scanning service.
+type Service interface {
+	apiGRPC.APIService
+
+	v1.NodeInventoryServiceServer
+}
+
+// NewService returns the service for node scanning
+func NewService(nodeName string) Service {
+	return &serviceImpl{
+		inventoryCollector: &nodeinventory.Scanner{},
+		nodeName:           nodeName,
+	}
+}
+
+type serviceImpl struct {
+	inventoryCollector *nodeinventory.Scanner
+	nodeName           string
+}
+
+func (s *serviceImpl) GetNodeInventory(ctx context.Context, req *v1.GetNodeInventoryRequest) (*v1.GetNodeInventoryResponse, error) {
+	inventoryScan, err := s.inventoryCollector.Scan(s.nodeName)
+	if err != nil {
+		log.Errorf("Error running inventoryCollector.Scan(%s): %v", s.nodeName, err)
+		return nil, errors.New("Internal scanner error: failed to scan node")
```

Comment:
I think the process in which we build an inventory is called "node analysis"  in the scanner team. We also call it inventorying but it is something introduced only recently.

Reply:
Is there a suggestion here? Should it say `Internal scanner error: failed to analyze node` instead?

Reply:
I think either would work. 
Changing to `analyze` would make it more consistent with the language the scanner team uses.
But as this error would eventually make its way back to Compliance as a return, it is likely to be shown to users. 
Therefore, maybe it would be the right call to stick to `scan`, as we call this feature `node scanning` in almost all user facing log messages.
I would tend to lean towards keeping it as `scan`, but that's non-blocking for me.

Reply:
Options like:
- failed to analyze node
- failed to build node inventory

would be okay for me. 

I think Scanner team uses additionally terms like: analyzing (scanning disk to search for pkgs), enriching (adding vulnerabilities data to the inventory). We never fully synced our dictionaries, so I guess we can also keep it the way it is.